vspc/server.py

```diff
@@ -81,9 +83,74 @@
                   DO_PROXY + WILL_PROXY + WONT_PROXY)
 
 
-class VspcServer(object):
+class BackgroundWriter(threading.Thread):
+    WRITE_TRIGGERING_DATA_AMOUNT = 4096
+
+    def __init__(self, write_queues):
+        self._write_queues = write_queues
+        # the event we set to trigger the background thread write
+        self._writes_available = threading.Event()
+        self._data_available_estimate = 0
+        threading.Thread.__init__(self)
+        self._should_stop = False
+        self._serial_log_dir = Path(CONF.serial_log_dir)
+
+    def stop(self):
+        self._should_stop = True
+        self._writes_available.set()
+
+    def set_writes_available(self, count=0, force=False):
```

Comment:
I think `count` is a bad name for this parameter because it's very non-descript. How about

```suggestion
    def set_writes_available(self, new_data_len=0, force=False):
```
?

Reply:
`new_data_len` sounded to me like it would be the new data length, not that it has to be added hm ... not sure if that's just me. What do you think about `data_len` instead?

Reply:
`data_len` is also fine. the `new` suggested to me that this is the length to be added, from incoming, i.e. "new" data. but data_len is already 90% better, so i don't mind the difference.

Reply:
I've named it `data_len`.